src/routes/intro/IntroContainer.js

```diff
@@ -0,0 +1,184 @@
+    }, {
+      key: t('offers'),
+      title: t('offers'),
+      description: t('offersDescription'),
+      renderContent: this.renderImageContent(Offers)
+    }, {
+      key: t('languageChange'),
+      title: t('languageChange'),
+      description: t('languageChangeDescription'),
+      renderContent: this.renderImageContent(Language)
+    }, {
+      key: t('inquiry'),
+      title: t('inquiryTitle'),
+      renderContent: this.renderSettings
+    }]
+  }
+
+  setAllowPushNotifications = () => this.setState(prevState =>
+    ({ allowPushNotifications: !prevState.allowPushNotifications }))
+
+  setUseLocationAccess = () => this.setState(prevState => ({ useLocationAccess: !prevState.useLocationAccess }))
+
+  setAllowSentry = () => this.setState(prevState => ({ allowSentry: !prevState.allowSentry }))
+
+  renderSettings = (): React.Node => {
+    const { t, theme } = this.props
+    const themeColor = theme.colors.themeColor
+    const { allowPushNotifications, useLocationAccess } = this.state
+
+    return <>
+      <SettingItem bigTitle title={t('pushNewsTitle')} description={t('pushNewsDescription')}
+                   onPress={this.setAllowPushNotifications} theme={theme}>
+        <Switch thumbColor={themeColor} trackColor={{ true: themeColor }}
+                onValueChange={this.setAllowPushNotifications} value={allowPushNotifications} />
+      </SettingItem>
+      <SettingItem bigTitle title={t('locationTitle')} description={t('locationDescription')}
+                   onPress={this.setUseLocationAccess} theme={theme}>
+        <Switch thumbColor={themeColor} trackColor={{ true: themeColor }}
+                onValueChange={this.setUseLocationAccess} value={useLocationAccess} />
+      </SettingItem>
+      <SettingItem bigTitle title={t('sentryTitle')} description={t('sentryDescription')}
+                   onPress={this.setAllowSentry} theme={theme}>
+        <Switch thumbColor={themeColor} trackColor={{ true: themeColor }}
+                onValueChange={this.setAllowSentry} value={useLocationAccess} />
+      </SettingItem>
+    </>
+  }
+
+  onAccept = () => { this.onDone(true) }
+
+  onContinue = () => { this.onDone(false) }
+
+  onDone = async (accept: boolean) => {
+    const allowPushNotifications = accept ? true : this.state.allowPushNotifications
+    const errorTracking = accept ? true : this.state.allowSentry
+    const useLocationAccess = accept ? true : this.state.useLocationAccess
+
+    if (errorTracking) {
+      const sentry = new SentryIntegration()
+      await sentry.install()
+    }
+
+    if (useLocationAccess) {
+      // TODO request permission, return if not granted
+    }
+
+    await this._appSettings.setSettings({ errorTracking, allowPushNotifications, useLocationAccess })
+    this._appSettings.setIntroShown()
+    this.props.navigation.navigate('Landing')
+  }
+
+  goToSlide = (index: number) => {
+    if (!this._flatList.current) {
+      throw Error('ref not correctly set')
```

Comment:
Maybe it makes sense not to treat this as an error but just ignore the call of goToSlide.

Reply:
I don't think so. This should never be the case, but if, it is because of an implementation error, in which case an error makes way more sense imo.